clientd3d/music.c

```diff
@@ -146,22 +153,60 @@ DWORD OpenMidiFile(const char *lpszMIDIFileName)
 
    sprintf(filename, "%s%s", current_dir, lpszMIDIFileName);
    debug(("music filename = %s\n", filename));
-
-   memset(&mciOpenParms, 0, sizeof(MCI_OPEN_PARMS));
-   mciOpenParms.lpstrDeviceType = "sequencer";
-   mciOpenParms.lpstrElementName = filename;
-   if (dwReturn = mciSendCommand(0, MCI_OPEN, MCI_OPEN_ELEMENT,
+   // Is it a background music or jala music file?
+   if (case_bg)
```

Comment:
This is how I had it working about 10 days ago.  The reason for the change: Loading a midi file into memory eats up a bunch of compute power.  Its probably not an issue on modern machines but my testing environment is a Gen 3 or 4 intel I5 Dell laptop.  For me I get at least 250ms of lag while the new midi is loaded into memory.  The result is that if I am singing and start running which breaks my trance, the player gets stuck for a good chunk of time before he can run away.

The old version with only one device did work.  I've got it saved on another branch just in case.

Reply:
I've greatly updated this section - thanks for the feedback.